src/new-storage/DeleteExpiredCmd.js

```diff
@@ -0,0 +1,139 @@
+                const url = `${this.baseUrl}/api/v1/streams/${stream.streamId}/validation`
+                return fetch(url).then((res) => res.json()).then((json) => {
+                    return {
+                        streamId: stream.streamId,
+                        partition: stream.partition,
+                        storageDays: parseInt(json.storageDays)
+                    }
+                }).catch((e) => console.error(e))
+            })
+        })
+
+        return Promise.all(tasks)
+    }
+
+    async _deleteExpired(streamsInfo) {
+        const tasks = streamsInfo.map((stream) => {
+            const { bucketId, dateCreate, streamId, partition } = stream
+            const queries = [
+                {
+                    query: 'DELETE FROM bucket WHERE stream_id = ? AND partition = ? AND date_create = ?',
+                    params: [streamId, partition, dateCreate]
+                },
+                {
+                    query: 'DELETE FROM stream_data WHERE stream_id = ? AND partition = ? AND bucket_id = ?',
+                    params: [streamId, partition, bucketId]
+                }
+            ]
+
+            return this.limit(async () => {
+                await this.cassandraClient.batch(queries, {
+                    prepare: true
+                }).catch((err) => {
+                    console.error(err)
+                })
+            })
+        })
+
+        return Promise.all(tasks)
+    }
+
+    async _getExpiredBuckets(streamsInfo) {
+        const result = []
+
+        const query = 'SELECT * FROM bucket WHERE stream_id = ? AND partition = ? AND date_create <= ?'
+
+        const tasks = streamsInfo.map((stream) => {
+            const { streamId, partition, storageDays } = stream
+            const timestampBefore = Date.now() - 1000 * 60 * 60 * 24 * storageDays
+            const params = [streamId, partition, timestampBefore]
+
+            return this.limit(async () => {
+                const resultSet = await this.cassandraClient.execute(query, params, {
+                    prepare: true,
+                }).catch((e) => console.error(e))
+
+                resultSet.rows.forEach((row) => {
+                    result.push({
+                        bucketId: row.id,
+                        dateCreate: row.date_create,
+                        streamId: row.stream_id,
+                        partition: row.partition,
+                        storageDays
+                    })
+                })
+            })
+        })
+
+        await Promise.all(tasks)
+        return result
+    }
+
+    async run() {
+        try {
+            const streams = await this._getStreams()
+            console.info(`Found ${streams.length} unique streams`)
+
+            const streamsInfo = await this._fetchStreamsInfo(streams)
+            const expiredBuckets = await this._getExpiredBuckets(streamsInfo)
+
+            console.info(`Found ${expiredBuckets.length} expired buckets`)
```

Comment:
unless you filter out errors, if an `expiredBuckets` request fails you'll be counting failures here too